cmd/k8s-operator/sts.go

```diff
@@ -831,9 +839,15 @@ func tailscaledConfig(stsC *tailscaleSTSConfig, newAuthkey string, oldSecret *co
 		}
 		conf.AuthKey = key
 	}
+
 	capVerConfigs := make(map[tailcfg.CapabilityVersion]ipn.ConfigVAlpha)
+	capVerConfigs[107] = *conf
+
+	// AppConnector config option is only understood by clients of capver 107 and newer.
+	conf.AppConnector = nil
```

Comment:
Here is an edge case where if someone is attempting to create an app connector via Connector, but are using an older proxy version (pre-capver107) and they set `connector.spec.appConnector.routes`, the app connector gets removed from tailscaled config, but not the routes, so they will essentially create a subnet router for those routes. I think it's ok to leave that as is, because that is a broken state anyway